src/test/scala/com/vngrs/etl/transformers/SqlTransformSpec.scala

```diff
@@ -0,0 +1,143 @@
+package com.vngrs.etl.transformers
+
+import com.vngrs.etl.SparkSpec
+import com.vngrs.etl.exceptions.NoSchemaException
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.expressions.{GenericRow, GenericRowWithSchema}
+import org.apache.spark.sql.types.{DataTypes, StructField, StructType}
+
+/**
+  * SQL Transformer Specs
+  */
+// Following wart does not work with scalatest's intercept functionality
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+class SqlTransformSpec extends SparkSpec {
+
+  "An SQL Transformer" should "select specified columns" in {
+    val data = generateRowData()
+
+    val transformer = SqlTransform(s"SELECT name, surname FROM ${SqlTransform.defaultTableName}")
+
+    val transformedRdd = transformer(data)
+
+    transformedRdd.first().schema.fields.map(_.name) should equal (Array("name", "surname"))
```

Comment:
We use expressions like `.first().schema.fields.map(_.name)` quite often in our tests. I bet scalatest matchers make it shorter.